sisl/geometry.py

```diff
@@ -2881,6 +2882,10 @@ def __plot__(self, axis=None, supercell=True, axes=False, *args, **kwargs):
         else:
             axes.scatter(xyz[:, axis[0]], xyz[:, axis[1]], s=area, c=colors, alpha=0.8)
 
+        if atom_indices:
```

Comment:
It should be move up into the `else` clause

Secondly, it needs to be duplicated for the 3D plot case.